var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -0,0 +1,69 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Hiop(CMakePackage):
+    """HiOp is an optimization solver for solving certain mathematical
+    optimization problems expressed as nonlinear programming problems.
+    HiOp is a lightweight HPC solver that leverages application's existing
+    data parallelism to parallelize the optimization iterations by using
+    specialized linear algebra kernels."""
+
+    homepage = "https://github.com/LLNL/hiop"
+    url      = "https://github.com/LLNL/hiop"
+
+    version('0.1', git='https://github.com/LLNL/hiop',
+            tag='v0.1')
+
+    variant('mpi', default=True,
+            description='Enable/Disable MPI')
+
+    variant('deepchecking', default=True,
+            description='Ultra safety checks - \
+            used for increased robustness and self-diagnostics')
+
+    depends_on('mpi', when='+mpi')
+    depends_on('lapack')
+    depends_on('blas')
+
+    flag_handler = CMakePackage.build_system_flags
+
+    def cmake_args(self):
+        args = []
+        spec = self.spec
+
+        if '+mpi' in spec:
+            args.append("-DCMAKE_CXX_COMPILER=%s" % spec['mpi'].mpicxx)
```

Comment:
The `setup_environment` method for every spack package that has the `provides('mpi')` directive should set the environment variables needed for CMake to detect the MPI compiler, so you shouldn’t need this line.

Reply:
I agree.  I'll make this change.